src/targets/gpu/convolution.cpp

```diff
@@ -121,6 +152,69 @@ shape miopen_convolution::find(context& ctx, const shape& output_shape, std::vec
         MIGRAPHX_THROW("MIOpen Convolution: find convolution failed");
     algo = perf.fwd_algo;
 
+    if(enabled(MIGRAPHX_MIOPEN_FIND_2{}))
+    {
+        miopenProblem_t conv_problem;
+        status = miopenCreateConvProblem(&conv_problem, cd.get(), miopenProblemDirectionForward);
+        if(status != miopenStatusSuccess)
+        {
+            MIGRAPHX_THROW("conv problem creation failed");
+        }
+        auto set_tensor_descriptor = [conv_problem](miopenTensorArgumentId_t name,
+                                                    tensor_descriptor& desc) {
+            auto status = miopenSetProblemTensorDescriptor(conv_problem, name, desc.get());
+            if(status != miopenStatusSuccess)
+            {
+                MIGRAPHX_THROW("setting problem tensor description failed");
+            }
+        };
+
+        set_tensor_descriptor(miopenTensorConvolutionX, x_desc);
+        set_tensor_descriptor(miopenTensorConvolutionW, w_desc);
+        set_tensor_descriptor(miopenTensorConvolutionY, y_desc);
+
+        auto solutions    = std::vector<miopenSolution_t>{};
+        std::size_t found = 0;
+        solutions.resize(100);
+
+        // TODO: create managed pointer for find_options
+        miopenFindOptions_t options;
+
+        // add checks for return statues  later
+        status = miopenCreateFindOptions(&options);
+        // not sure which tuning option to select
+        status = miopenSetFindOptionTuning(options, 1);
+        // not needed by default it is ordered by time
+        status = miopenSetFindOptionResultsOrder(options, miopenFindResultsOrderByTime);
+        // not sure if i should be using workspace size from GetWorkSpaceSize() call from earlier
+        status = miopenSetFindOptionWorkspaceLimit(options, workspace_size);
```

Comment:
Dont set this either, we dont want to limit workspace.

Reply:
Removed.